var/spack/repos/builtin/packages/yambo/package.py

```diff
@@ -0,0 +1,162 @@
+    variant(
+        'profile',
+        values=('time', 'memory'),
+        default='',
+        description='Activate profiling of specific sections',
+        multi=True
+    )
+
+    variant(
+        'io',
+        values=('iotk', 'etsf-io'),
+        default='',
+        description='Activate support for different io formats (requires network access)',  # noqa
+        multi=True
+    )
+
+    # MPI + OpenMP parallelism
+    variant('mpi', default=True, description='Enable MPI support')
+    variant('openmp', default=False, description='Enable OpenMP support')
+
+    depends_on('blas')
+    depends_on('lapack')
+
+    # MPI dependencies are forced, until we have proper forwarding of variants
+    depends_on('mpi', when='+mpi')
+    depends_on('netcdf+mpi', when='+mpi')
+    depends_on('hdf5+mpi', when='+mpi')
+    depends_on('fftw+mpi', when='+mpi')
+    depends_on('scalapack', when='+mpi')
+
+    depends_on('netcdf~mpi', when='~mpi')
+    depends_on('hdf5~mpi', when='~mpi')
+    depends_on('fftw~mpi', when='~mpi')
+
+    depends_on('hdf5+fortran')
+    depends_on('netcdf-fortran')
+    depends_on('libxc@2.0.3:')
+
+    build_targets = ['all']
+
+    parallel = False
+
+    # The configure in the package has the string 'cat config/report'
+    # hard-coded, which causes a failure at configure time due to the
+    # current working directory in Spack. Fix this by using the absolute
+    # path to the file.
+    @run_before('configure')
+    def filter_configure(self):
+        report_abspath = join_path(self.build_directory, 'config', 'report')
+        filter_file('config/report', report_abspath, 'configure')
+
+    def enable_or_disable_time(self, activated):
+        return '--enable-time-profile' if activated else '--disable-time-profile'  # noqa: E501
+
+    def enable_or_disable_memory(self, activated):
+        return '--enable-memory-profile' if activated else '--disable-memory-profile'  # noqa: E501
+
+    def enable_or_disable_openmp(self, activated):
+        return '--enable-open-mp' if activated else '--disable-open-mp'
+
+    def configure_args(self):
+
+        args = [
+            # As of version 4.2.1 there are hard-coded paths that make
+            # the build process fail if the target prefix is not the
+            # configure directory
+            '--prefix={0}'.format(self.stage.source_path),
+            '--disable-keep-objects',
+            '--with-editor=none'
+        ]
+        spec = self.spec
+
+        # Double precision
+        args.extend(self.enable_or_disable('dp'))
+
+        # Application profiling
+        args.extend(self.enable_or_disable('profile'))
+
+        # MPI + threading
+        args.extend(self.enable_or_disable('mpi'))
+        args.extend(self.enable_or_disable('openmp'))
+
+        # LAPACK
+        if '+mpi' in spec:
+            args.append('--with-scalapack-libs={0}'.format(
+                spec['scalapack'].libs + spec['lapack'].libs
```

Comment:
you may want to add `blas` here, just in case `lapack` and `blas` are indeed split into different libs by provider.